### PR TITLE
Feat/HNG 113 create related articles on dynamic blog page

### LINKED ISSUE #113: [FEATURE] Create and implement the API integration for Related Articles on the dynamic blog page
# **Description**
Implement the design for the related articles, make it responsive on both the web and mobile screens. Make use of reusable components for the related ariticles. Integrate the api to map out the all related articles in the blog details page

# **Acceptance Criteria**
- Pixel perfect implementaiton
- Responsive implementation
- Clean code
- Make use of reusable components where needed, make one if needed
- Follow the [contribution guidelines](https://github.com/hngprojects/hng_boilerplate_nextjs/blob/0acf613964ea064ae211a91d0a74d13938e4bc23/CONTRIBUTING.md) and [code of conduct](https://github.com/hngprojects/hng_boilerplate_nextjs/blob/0acf613964ea064ae211a91d0a74d13938e4bc23/CODE_OF_CONDUCT.md)

# **Requirements**
- Display three related articles
- Proper error handling
- Proper loading indicator
- None breaking api integration

# **Testing**
- Write test to ensure the page renders the required contents

# **Deadline**
Friday 11:59 pm WAT

# **Tasks**
- [ ] Implement all the sections in the design
- [ ] Responsiveness across all screens
- [ ] Make use of reusable components where needed
- [ ] Write test for the implementations
- [ ] Proper commit message

# **Images**
Frame 53126
Frame 1171275100

[Figma design](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=1632-11446&m=dev)

### PR DESCRIPTION
**Pull request by Kored28**
# Description

**Closes #113**

# Changes proposed

## What were you told to do?
Implement the design for the related article in the dynamic blog detail page

[FEATURE] Create and implement the API integration for Related Articles on the dynamic blog page

## What did you do?

- I implemented all the sections in the design
- I made the component responsive
- I made use of reusable components where needed
- I wrote test for the implementations
- I made sure I used proper commit message

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos
Design
Screenshot (8)
Screenshot (9)

Implementation
Screenshot (6)
Screenshot (7)

Testing
Screenshot (11)
Screenshot (12)

Proof the test worked
Screenshot (16)

### CODE REVIEW

**Comment by Kored28** [comment]
There was no custom color that matched the color in the design for Blog Card title and loading

File: tailwind.config.ts
Comment (JulianaSau) [inline tailwind.config.ts:1]: These changes seem very unnecessary. remove them

File: src/test/Button.test.tsx
Comment (JulianaSau) [inline src/test/Button.test.tsx:1]: same to the changes here

File: src/components/blog/BlogCard.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      <div className="flex h-[147px] w-[372px] flex-col gap-[24px] p-[16px] md:h-[137px] md:w-[354px]">
+        <p className="w-[320px] text-[20px] font-bold text-[#525252]">
+          {newsType.title}
+        </p>
+        <div className="flex items-center justify-between">
+          <p className="text-[14px] font-normal text-[#525252]">
+            {newsType.date}
+          </p>
+          <div className="rounded-[16px] bg-[#F3F4F6] p-[8px]">
+            <p className="text-[14px] font-normal text-[#525252]">
```
Comment (JulianaSau) [inline src/components/blog/BlogCard.tsx:52]: for the colors, if they dont exist, add them to the tailwind config. dont use custom colors

File: src/app/globals.css
Comment (JulianaSau) [inline src/app/globals.css:1]: these changes seem unnecessary

File: src/app/favicon.ico
Comment (JulianaSau) [inline src/app/favicon.ico:1]: why did you delete this?

File: src/app/blog/Blog.tsx
Comment (JulianaSau) [inline src/app/blog/Blog.tsx:1]: what does this page do?
Doesnt seem to be used anywhere

**Review by JulianaSau — CHANGES_REQUESTED** [review]

File: src/app/globals.css
Comment (Kored28) [reply src/app/globals.css:1]: It was what I pulled from upstream yesterday

File: src/components/blog/BlogCard.tsx (hunk 1 above)
Comment (Kored28) [reply src/components/blog/BlogCard.tsx:52]: Okay

File: src/app/favicon.ico
Comment (Kored28) [reply src/app/favicon.ico:1]: It was what I pulled from upstream yesterday

File: src/app/blog/Blog.tsx
Comment (Kored28) [reply src/app/blog/Blog.tsx:1]: I used it so I could see what I was doing on the local Host. I will remove it

File: tailwind.config.ts
Comment (Kored28) [reply tailwind.config.ts:1]: It is because of the pull from stream I made yesterday

**Pull request closed without merging**